packages/@n8n/task-runner-python/src/task_analyzer.py

```diff
@@ -0,0 +1,205 @@
+import ast
+import hashlib
+import sys
+from typing import Set, Tuple
+from collections import OrderedDict
+
+from src.errors import SecurityViolationError
+from src.constants import MAX_VALIDATION_CACHE_SIZE
+
+CacheKey = Tuple[str, Tuple]  # (code_hash, allowlists_tuple)
+CachedViolations = list[str]
+ValidationCache = OrderedDict[CacheKey, CachedViolations]
+
+ERROR_RELATIVE_IMPORT = "Relative imports are disallowed."
+ERROR_DYNAMIC_IMPORT = "Dynamic imports using `__import__()` with variables are disallowed, because they can bypass allowlist validation at runtime."
+ERROR_DYNAMIC_IMPORTLIB = "Dynamic imports using `importlib` with variables are disallowed, because they can bypass allowlist validation at runtime."
+ERROR_STDLIB_DISALLOWED = "Import of standard library module '{module}' is disallowed. Allowed stdlib modules: {allowed}"
+ERROR_EXTERNAL_DISALLOWED = "Import of external package '{module}' is disallowed. Allowed external packages: {allowed}"
+ERROR_SECURITY_VIOLATIONS = "Security violations detected:\n{violations}"
+
+
+class ImportValidator(ast.NodeVisitor):
```

Comment:
Would be nice to have a description for what this does, as well as unit tests

Reply:
> as well as unit tests

I'm deferring test coverage until after we bring this out. See [here](https://n8nio.slack.com/archives/C0995C3KLUF/p1756117697844679?thread_ts=1756116420.613699&cid=C0995C3KLUF).